drivers/adc/adc_stm32_v2.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2019 Song Qiang <songqiang1304521@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * For F1/F2/F4/F7 series stm32 support.
+ */
+
+#include "adc_stm32.h"
+
+void adc_stm32_init_soc(ADC_TypeDef *adc)
+{
+	adc->CR2 |= ADC_CR2_ADON;
```

Comment:
Why not using `LL_ADC_Enable` and factorize this part?

Reply:
LL_ADC_Enable can be applied here, but I was thinking if I used LL lib somewhere, it's better to use LL lib everywhere. We talked about why I do not think LL lib is good for this generic driver, and except this function, some functions like getting EOC flags from registers uses different functions between different series of stm32 in LL lib.
So should I use LL and direct register access in the same driver?
I personally prefer to use direct register access for all operations.